src/services/pairing/utils.ts

```diff
@@ -0,0 +1,22 @@
+import WalletConnect from '@walletconnect/client'
+
+export const formatPairingUri = (wcUri: string) => {
+  const PAIRING_MODULE_URI_PREFIX = 'safe-'
+
+  // A disconnected session returns URI with an empty `key`
+  if (!wcUri || !/key=.+/.test(wcUri)) {
+    return
+  }
+
+  return `${PAIRING_MODULE_URI_PREFIX}${wcUri}`
+}
+
+export const killPairingSession = (connector: InstanceType<typeof WalletConnect>) => {
+  // WalletConnect throws if no `peerId` is sett when attempting to `killSession`
+  // We therefore manually set it in order to `killSession` without throwing
+  if (!connector.peerId) {
+    connector.peerId = '_willKill'
```

Comment:
Is `_willKill` some kind of constant from WC we have to use or random so that its not empty? It seems a bit like a workaround because I would assume that if `connector.connected` that there always is a `peerId` which is more intuitive i.e. "only kill session if we have one"

Reply:
No, `"_willKill"` is something arbitrary. It is a workaround. `connector.connected` is `true` when a session is created but `peerId` is only populated when a peer connects. Should I choose something other than `"_willKill"` or extend the comment?